### PR TITLE
Remaining perpetual rework

### PR DESCRIPTION
**Pull request by Northmoc**
new card "Legion Reconsecrator" is in this batch

### CODE REVIEW

File: forge-game/src/main/java/forge/game/ability/effects/AnimateAllEffect.java [hunk 1]
```diff
@@ -133,15 +132,16 @@ public void resolve(final SpellAbility sa) {
...
+            ZoneType.listValueOf("Battlefield");
 
         if (sa.usesTargeting() || sa.hasParam("Defined")) {
             list = getTargetPlayers(sa).getCardsIn(z);
         } else {
             list = game.getCardsIn(z);
         }
 
-        list = CardLists.getValidCards(list, valid, sa.getActivatingPlayer(), host, sa);
+        list = AbilityUtils.filterListByType(list, valid, sa);
```
Comment (Northmoc) [inline forge-game/src/main/java/forge/game/ability/effects/AnimateAllEffect.java:144]: I made this like PumpAllEffect – with this, Teferi's Contingency works. Is this OK?

File: forge-gui/res/cardsfolder/n/niambi_beloved_protector.txt [hunk 2]
```diff
@@ -4,11 +4,11 @@ Types:Legendary Creature Human Cleric
...
-SVar:PerpetualBecomesTargetDraw:Mode$ Continuous | Affected$ Card.IsRemembered | AffectedZone$ Battlefield,Hand,Graveyard,Exile,Stack,Library,Command | AddTrigger$ BecomesTarget | Description$ Whenever this creature becomes the target of a spell or ability, draw a card. This ability triggers only once each turn.
-SVar:BecomesTarget:Mode$ BecomesTarget | ValidTarget$ Card.Self | ActivationLimit$ 1 | TriggerZones$ Battlefield | Execute$ TrigDraw
+SVar:TrigChangeZone:DB$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ValidTgts$ Creature.YouOwn+nonLegendary+ThisTurnEnteredFrom_Battlefield | RememberChanged$ True | TgtPrompt$ Choose target nonlegendary creature card in your graveyard that was put there from the battlefield this turn | ChangeNum$ 1 | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ Remembered | Triggers$ BecomesTarget | Duration$ Perpetual | SubAbility$ DBCleanup
+SVar:BecomesTarget:Mode$ BecomesTarget | ValidTarget$ Card.Self | ActivationLimit$ 1 | TriggerZones$ Battlefield | Execute$ TrigDraw | TriggerDescription$ Whenever this creature becomes the target of a spell or ability, draw a card. This ability triggers only once each turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/n/niambi_beloved_protector.txt:9]: not sure this TriggerZones is correct

Comment (Northmoc) [reply forge-gui/res/cardsfolder/n/niambi_beloved_protector.txt:9]: Guess it is correct... Simisays tested in Arena

File: forge-game/src/main/java/forge/game/ability/effects/AnimateEffectBase.java [hunk 3]
```diff
@@ -127,7 +127,18 @@ public static void doAnimate(final Card c, final SpellAbility sa, final Integer
             c.addHiddenExtrinsicKeywords(timestamp, 0, hiddenKeywords);
         }
 
-        c.addColor(colors, !sa.hasParam("OverwriteColors"), timestamp, 0, false);
+        if (!colors.equals(ColorSet.getNullColor())) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/AnimateEffectBase.java:130]: you just broke _Ancient Kavu_
probably the default value should be null

**Review by tool4ever — APPROVED** [review]

**Comment by Northmoc** [comment]
If you want we could merge this one and do more in a new PR? I'll make it not a draft

**Comment by tool4ever** [comment]
that was the plan :P

**Pull request merged**